test-data/unit/check-classes.test

```diff
@@ -2617,6 +2617,27 @@ b = a.bar
 [out]
 main:9: error: Incompatible types in assignment (expression has type "A", variable has type "B")
 
+[case testDecoratedGetAttribute]
+from typing import Callable, TypeVar
+
+T = TypeVar('T', bound=Callable)
+
+def decorator(f: T) -> T:
+    return f
+
+a, b = None, None # type: A, B
+class A:
+    @decorator
+    def __getattribute__(self, x: str) -> A:
+        return A()
+class B: pass
+
+a = a.foo
+b = a.bar
+[builtins fixtures/tuple.pyi]
+[out]
+main:16: error: Incompatible types in assignment (expression has type "A", variable has type "B")
```

Comment:
Style nit: test cases will be easier to read and maintain if messages are introduced using `# E: <message:` (or `# N: ...`).